cmd/ipfs/init.go

```diff
@@ -85,29 +78,44 @@ IPFS and are now interfacing with the ipfs merkledag!
 For a short demo of what you can do, enter 'ipfs tour'
 `
 
-func initWithDefaults(configRoot string) error {
-	_, err := doInit(configRoot, "", false, nBitsForKeypairDefault)
+func initWithDefaults(repoRoot string) error {
+	_, err := doInit(repoRoot, false, nBitsForKeypairDefault)
 	return debugerror.Wrap(err)
 }
 
-func doInit(configRoot string, dspathOverride string, force bool, nBitsForKeypair int) (interface{}, error) {
+func doInit(repoRoot string, force bool, nBitsForKeypair int) (interface{}, error) {
 
-	u.POut("initializing ipfs node at %s\n", configRoot)
+	u.POut("initializing ipfs node at %s\n", repoRoot)
 
-	configFilename, err := config.Filename(configRoot)
-	if err != nil {
-		return nil, debugerror.New("Couldn't get home directory path")
-	}
-
-	if u.FileExists(configFilename) && !force {
+	if fsrepo.IsInitialized(repoRoot) && !force {
 		return nil, errCannotInitConfigExists
 	}
 
-	conf, err := initConfig(configFilename, dspathOverride, nBitsForKeypair)
+	conf, err := initConfig(nBitsForKeypair)
 	if err != nil {
 		return nil, err
 	}
 
+	if !fsrepo.IsInitialized(repoRoot) {
+		if err := fsrepo.Init(repoRoot, conf); err != nil {
+			return nil, err
+		}
+	} else {
+		r := fsrepo.At(repoRoot)
+		if err := r.Open(); err != nil {
+			return nil, err
+		}
+		if err := r.SetConfig(conf); err != nil {
+			return nil, err
+		}
+		if err := r.Close(); err != nil {
+			return nil, err
+		}
```

Comment:
if i'm reading this right, `ipfs init -f` would not blow out the datastore, correct?

I wonder which is easier on the user:
```sh
ipfs init -f   # re-inits config
rm -rf ~/.go-ipfs/datastore

ipfs init -f   # re-inits config, blows out everything
ipfs init --config  # only reinits config

# no ipfs init -f
ipfs init --config  # only reinits config
ipfs init --datastore  # only reinits datastore
```